crates/jstzd/tests/octez_baker_test.rs

```diff
@@ -0,0 +1,136 @@
+    tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
+    let current_level = get_block_level(&rpc_endpoint.to_string()).await;
+    assert_eq!(last_level, current_level);
+    let _ = octez_node.kill().await;
+}
+
+async fn spawn_octez_node() -> (octez_node::OctezNode, TempDir) {
+    let temp_dir = TempDir::new().unwrap();
+    let data_dir = temp_dir.path();
+    let mut config_builder = OctezNodeConfigBuilder::new();
+    let run_options = OctezNodeRunOptionsBuilder::new()
+        .set_synchronisation_threshold(0)
+        .build();
+    config_builder
+        .set_binary_path("octez-node")
+        .set_network("sandbox")
+        .set_data_dir(data_dir.to_str().unwrap())
+        .set_run_options(&run_options);
+    let octez_node = octez_node::OctezNode::spawn(config_builder.build().unwrap())
+        .await
+        .unwrap();
+    let node_ready = retry(10, 1000, || async { octez_node.health_check().await }).await;
+    assert!(node_ready);
+    (octez_node, temp_dir)
+}
+
+async fn import_bootstrap_keys(octez_client: &OctezClient) {
+    for i in 1..5 {
+        let alias = format!("bootstrap{}", i);
+        let key = match i {
+            1 => "unencrypted:edsk3gUfUPyBSfrS9CCgmCiQsTCHGkviBDusMxDJstFtojtc1zcpsh",
+            2 => "unencrypted:edsk39qAm1fiMjgmPkw1EgQYkMzkJezLNewd7PLNHTkr6w9XA2zdfo",
+            3 => "unencrypted:edsk4ArLQgBTLWG5FJmnGnT689VKoqhXwmDPBuGx3z4cvwU9MmrPZZ",
+            4 => "unencrypted:edsk2uqQB9AY4FvioK2YMdfmyMrer5R8mGFyuaLLFfSRo8EoyNdht3",
+            _ => panic!("Invalid key"),
+        };
+        octez_client
+            .import_secret_key(&alias, key)
+            .await
+            .expect("Failed to generate bootstrap key");
+    }
+}
+
+async fn activate_alpha(octez_client: &OctezClient) {
+    // 3. import activator key
+    let activator = "activator".to_string();
+    octez_client
+        .import_secret_key(
+            &activator,
+            "unencrypted:edsk31vznjHSSpGExDMHYASz45VZqXN4DPxvsa4hAyY8dHM28cZzp6",
+        )
+        .await
+        .expect("Failed to generate activator key");
+    // 4. activate the alpha protocol
+    let params_file =
+        Path::new(std::env!("CARGO_MANIFEST_DIR")).join("tests/sandbox-params.json");
+    let protocol_activated = octez_client
+        .activate_protocol(
+            "ProtoALphaALphaALphaALphaALphaALphaALphaALphaDdp3zK",
+            "0",
+            &activator,
+            &params_file,
+        )
+        .await;
+    assert!(protocol_activated.is_ok());
+}
+
+async fn get_block_level(rpc_endpoint: &str) -> i32 {
+    let blocks_head_endpoint =
+        format!("{}/chains/main/blocks/head", rpc_endpoint.to_owned());
+    let response = get_request(&blocks_head_endpoint).await;
+    extract_level(&response)
+}
+
+fn extract_level(input: &str) -> i32 {
+    // Create a regex to match "level": followed by a number
+    let re = Regex::new(r#""level":\s*(\d+)"#).unwrap();
```

Comment:
This works but it seems to an abuse of regex. What about leaving this to `serde_json`? This should give the level:
```rs
Value::from(&response).as_object()?.get("header")?.as_object()?.get("level")?.as_u64()?
```

Reply:
i thought of doing it but given it's just a test, i picked the shorter solution